src/events.rs

```diff
@@ -245,16 +251,22 @@ impl<T: System> EventsDecoder<T> {
                     let _topics = Vec::<T::Hash>::decode(input)?;
                     Raw::Event(event)
                 }
-                Err(Error::Runtime(err)) => Raw::Error(err),
                 Err(err) => return Err(err),
             };
 
-            r.push((phase, raw));
+            if event_errors.len() == 0 {
+                r.push((phase.clone(), raw));
+            }
+
+            for err in event_errors {
+                r.push((phase.clone(), Raw::Error(err)));
```

Comment:
```suggestion
                r.push((phase, Raw::Error(err)));
```

I guess the second clone is redundant

Reply:
The value is moved in the previous iteration of this loop.